entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityOperationService.kt

```diff
@@ -0,0 +1,116 @@
+package com.egm.stellio.entity.service
+
+import com.egm.stellio.entity.util.GraphBuilder
+import com.egm.stellio.entity.web.BatchEntityError
+import com.egm.stellio.entity.web.BatchOperationResult
+import com.egm.stellio.shared.model.BadRequestDataException
+import com.egm.stellio.shared.model.ExpandedEntity
+import com.egm.stellio.shared.util.NgsiLdParsingUtils.NGSILD_ENTITY_ID
+import com.egm.stellio.shared.util.NgsiLdParsingUtils.NGSILD_ENTITY_TYPE
+import com.google.common.graph.Graph
+import com.google.common.graph.Graphs.copyOf
+import org.springframework.stereotype.Component
+import kotlin.streams.toList
+
+/**
+ * Service to work on a list of entities.
+ */
+@Component
+class EntityOperationService(
+    private val entityService: EntityService,
+    private val graphBuilder: GraphBuilder
+) {
+
+    /**
+     * Splits [entities] by their existence in the DB.
+     */
+    fun splitEntitiesByExistence(entities: List<ExpandedEntity>): Pair<List<ExpandedEntity>, List<ExpandedEntity>> =
+        entities.partition {
+            entityService.exists(it.id)
```

Comment:
That will trigger a lot of requests (one per entity in the batch query). It could be optimized with a query using the IN operator (https://neo4j.com/docs/cypher-manual/current/clauses/where/#where-in-operator).

Reply:
I've commited an implementation using IN, I'm waiting for your feedback on this.